drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,371 @@
+        pool = multiprocessing.Pool(processes=config.capa.worker_pool_processes)
+        yield from pool.starmap(
+            static_capa_analysis, map(lambda dump: (dumps / dump, rules), malware_dumps)
+        )
+
+
+def format_capa_address(address: Union[Tuple, ca.Address]) -> Dict:
+    """This method formats capa address (in the format of tuples) into a single dictionary"""
+    if isinstance(address, tuple):
+        return functools.reduce(
+            lambda a, b: a | b, [format_capa_address(addr) for addr in address]
+        )
+    elif isinstance(
+        address,
+        (ca.AbsoluteVirtualAddress, ca.RelativeVirtualAddress, ca.FileOffsetAddress),
+    ):
+        return {"address": hex(address)}
+    elif isinstance(address, ca.DNTokenAddress):
+        return {"token": hex(address)}
+    elif isinstance(address, ca.DNTokenOffsetAddress):
+        return {
+            **format_capa_address(ca.DNTokenAddress(address.token)),
+            "offset": address.offset,
+        }
+    elif isinstance(address, ca.ProcessAddress):
+        return {"ppid": address.ppid, "pid": address.pid}
+    elif isinstance(address, ca.ThreadAddress):
+        # for the time being, we only collect the PID and PPID of a TTP
+        return {**format_capa_address(address.process)}
+    elif isinstance(address, ca.DynamicCallAddress):
+        # for the time being, we only collect the PID and PPID of a TTP
+        return {**format_capa_address(address.thread)}
+    elif isinstance(address, ca._NoAddress):
+        # empty address
+        return {}
+    else:
+        logger.debug("Encountered unknown address type: %s", type(address))
+        return {"address": address}
+
+
+def construct_ttp_block(
+    rule: Rule, addresses: List[Tuple[ca.Address, Result]]
+) -> Dict[str, Any]:
+    name = rule.name.split("/")[0]
+    mbc = rule.meta.get("mbc", None)
+    attck = rule.meta.get("att&ck", None)
+    occurrences = [format_capa_address(address=address) for address, _ in addresses]
+    occurrences = [
+        dict(t) for t in {tuple(d.items()) for d in occurrences}
+    ]  # remove duplicate addresses
+
+    ttp_block = dict()
+    ttp_block.update({"name": name})
+    ttp_block.update({"mbc": mbc} if mbc else {})
+    ttp_block.update({"att&ck": attck} if attck else {})
+    ttp_block.update({"occurrences": occurrences})
+    return ttp_block
+
+
+def construct_ttp_blocks(
+    rules: RuleSet,
+    capabilities_per_file: List[Tuple[Path, EngineMatchResults]],
+    filter_function=None,
+) -> Iterator[Dict[str, Any]]:
+    """construct a ttp block for each extracted capability"""
+    for _, capabilities in capabilities_per_file:
+        for name, addresses in capabilities.items():
+            if not filter_function or filter_function(rules[name]):
+                yield construct_ttp_block(rules[name], addresses)
+
+
+def capa_analysis(analysis_dir: Path) -> None:
+    config: DrakrunConfig = load_config()
```

Comment:
```suggestion
    config: DrakrunConfig = load_config()
```
This configuration object should be passed by caller instead of being loaded by the postprocessor itself, but currently PostprocessFunction doesn't offer such interface. To be changed in future.